src/kixi/gwyn/gwyn.clj

```diff
@@ -68,13 +71,48 @@
                                                coll-pumps-attending)}))))
         ds/dataset)})
 
+(defn adjust-avg-num-pumps
+  "Takes in the properties dataset. Adjust the average number of pumps
+   attending the fire depending on the value of the standard deviation."
+  [properties-data]
+  (-> properties-data
+      (wds/add-derived-column :adjusted-avg-pumps [:sd-pumps-attending :avg-pumps-attending]
+                              (fn [sd avg] (if (> sd 2) ;; have to chose what this value should be
+                                             (+ avg (u/safe-divide sd 2))
+                                             avg)))
+      (ds/select-columns [:property-type :num-fires :adjusted-avg-pumps :sd-pumps-attending])))
+
+(defn sort-by-pumps-and-fires
+  "Takes in the properties dataset with the average number of pumps adjusted.
+   Sort the property types by average number of pumps and by number of fires."
+  [adjusted-properties-data]
+  (->> (ds/row-maps adjusted-properties-data)
+       (sort-by (juxt :adjusted-avg-pumps :num-fires))
+       ds/dataset))
+
+(defn assign-generic-scores
+  "Takes in the properties dataset sorted by pumps and fires.
+   Use the sorting order to assign a score to each property type."
+  [sorted-properties-data]
+  (-> sorted-properties-data
+      (as-> data (ds/add-column data
```

Comment:
I find the `as->` here a little more complicated. I'd probably just lose it and pull the `(range 1 (inc (first (:shape data))))` out into a let (and also thread it).